US-1.1: Create Events

JimLau49 opened the pull request:
**Task Description**
This PR addresses #27 

Create events is now working (without image and location uploading) with validation for supported inputs. After entering valid form data, users will be able to send their events to our database.

**To run the E2E tests**
1. Make sure to run `npm i ` on the project to pickup the new dependencies 
1. Run the server `nx serve mull-api`
1. In the root directory run  `nx e2e mull-ui-e2e --watch` ( Cypress takes some time to open up)
**Note:** for `nx e2e mull-ui-e2e --watch` , the `--watch` option keeps Cypress open to watch for code changes instead of immediately running the tests and closing.
1. When the cypress window opens up select `create-event.spec.ts` and watch the tests go 🚀 
![image](https://user-images.githubusercontent.com/35787279/98070344-facd0080-1e2e-11eb-90e9-be2bf9ec04f3.png)

bit172 committed:
Add Create Event page #27

Co-authored-by: JimLau49 <jimmy71471@gmail.com>

bit172 committed:
Add Create Event page #27

Co-authored-by: JimLau49 <jimmy71471@gmail.com>

JimLau49 committed:
Rename project files #27

Co-authored-by: Jonathan <bit172@gmail.com>

JimLau49 committed:
Rename project files #27

Co-authored-by: Jonathan <bit172@gmail.com>

JimLau49 committed:
added pill options #27

Co-authored-by: Jonathan <bit172@gmail.com>

JimLau49 committed:
added pill options #27

Co-authored-by: Jonathan <bit172@gmail.com>

JimLau49 committed:
Fix styling for major components #27

    Co-authored-by: Jonathan <bit172@gmail.com>
    Co-authored-by: Cristian <aldea.cristian@outlook.com>

JimLau49 committed:
Fix styling for major components #27

Co-authored-by: Jonathan <bit172@gmail.com>
Co-authored-by: Cristian <aldea.cristian@outlook.com>

JimLau49 committed:
fix CSS of nav bar #27

Co-authored-by: Jonathan Hsu <bit172@gmail.com>
Co-authored-by: Ragith Saba <ragith.coolkid@gmail.com>

JimLau49 committed:
Add Calendar and time inputs  #27

Co-authored-by: Jonathan Hsu <bit172@gmail.com>
Co-authored-by: Ragith Saba <ragith.coolkid@gmail.com>

bit172 committed:
Refactor styling #27
Remove unecessary flex boxes
Add global styling for page containers
Made body styling more flexible for future pages

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>
Co-authored-by: Ragith Saba <ragith.coolkid@gmail.com>

bit172 committed:
Add formik and Yup to create event form #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Add Custom text input component #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Add validation for form fields #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Add event module, service and resolver #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Add Apollo Client to app #27
Add mutation to create event

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Add toast notifications, fix calendar #27
Fix variable names for event entity #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Moved time picker to component #27
Made toast update previous one when submitting event

bit172 committed:
Add unit tests for backend #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Add ui-lib unit tests #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Add front-end unit tests for create event #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Fix for travis and sonarcloud #27
Unused imports and default font family

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Add e2e tests for create event #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Fix backend bug #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Fix styling for create event #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

cristian-aldea commented on apps/mull-api/src/app/app.module.ts:
```diff
@@ -12,12 +12,15 @@ import { AppService } from './app.service';
 
 @Module({
   imports: [
+    UserModule,
+    EventModule,
     EntitiesModule,
     UserModule,
```
You import UserModule twice. Remove the import you added:
```suggestion
    EventModule,
    EntitiesModule,
    UserModule,
```

JimLau49 replied:
Really good suggestion on the selectors and the width :)

cristian-aldea commented on apps/mull-ui/src/app/app.scss:
```diff
@@ -1,16 +1,29 @@
+  @media (min-width: $breakpoint-mobile) {
+    padding: 7.5rem 0 1.25rem;
+  }
+}
```
Why should the body have padding and not some inner element, like your create event page? By doing it like this, you're forcing all other pages to have the same padding and space as the create event form, even when it wouldn't be what we want (like the camera for example).

For example, on desktop currently, the top padding for the body is larger than the height of the navbar, leaving some space that can't be removed
![Screenshot from 2020-11-04 09-52-04](https://user-images.githubusercontent.com/23544999/98126363-81abc880-1e83-11eb-85fb-cd385b99366a.png)

Same on mobile: 
![Screenshot from 2020-11-04 09-54-40](https://user-images.githubusercontent.com/23544999/98126649-d3545300-1e83-11eb-9820-97a383a35a8e.png)

I think this padding could be placed into the create event page, that way that page can control how it looks, and other pages can control how they look

cristian-aldea commented on apps/mull-ui/src/app/components/navigation-bar/navigation-bar.scss:
```diff
@@ -2,17 +2,20 @@
 
 .nav-container {
   background: $background-color;
```
Considering the changes you've done to how the navbar is styled (fixed position over the content and the content needs padding to avoid overlap), id recommends you set a fixed height for the navbar and then set the same value for the top padding (to make things easier you can make it a variable). 

```suggestion
  box-sizing: border-box;
  height: $nav-bar-height-mobile;
  background: $background-color;
  @media (min-width: $breakpoint-mobile) {
      height: $nav-bar-height-desktop;
  }
```

cristian-aldea commented on apps/mull-ui/src/app/pages/create-event/create-event.scss:
```diff
@@ -0,0 +1,90 @@
+
+.create-event-text {
+  color: $primary-color;
+}
```
I think this text is a bit small on my screen (especially on desktop) compared to the mockup and could use a margin on the bottom.
![image](https://user-images.githubusercontent.com/23544999/98150466-b415ef80-1e9c-11eb-8061-7a29afc7e45a.png)
![image](https://user-images.githubusercontent.com/23544999/98150480-bb3cfd80-1e9c-11eb-931e-a7562bd432ae.png)

cristian-aldea replied:
This comment was not addressed or replied to. I'll resolve it and make another comment with a suggestion

cristian-aldea commented on apps/mull-ui/src/app/pages/create-event/date-calendar/date-calendar.scss:
```diff
@@ -0,0 +1,27 @@
+
+.date-calendar {
+  display: flex;
+  justify-content: center;
```
Like @JimLau49  mentioned many times, using flex might not always be necessary, you can center this by using margin auto.

```suggestion
  margin: auto;
```

cristian-aldea commented on apps/mull-ui/src/app/pages/create-event/create-event.tsx:
```diff
@@ -0,0 +1,205 @@
+          {imageFile ? (
+            <img src={imageFile} style={{ width: '50%', height: '50%' }} alt="sup" />
+          ) : (
+            '🧼'
```
This soap icon doesn't match what is on the mockup

cristian-aldea commented on apps/mull-ui/src/app/pages/create-event/date-calendar/date-calendar.scss:
```diff
@@ -0,0 +1,27 @@
+.nice-dates-day:after {
+  background-color: $primary-color;
+  z-index: 1;
+}
```
You can add some more styling to make the calendar better match the mockup


```suggestion
}

.nice-dates-week-header {
  font-weight: bold;
}

.nice-dates-week-header_day {
  color: black;
}

```

Here's the result of the suggestion:
![image](https://user-images.githubusercontent.com/23544999/98149856-c3486d80-1e9b-11eb-8050-ff3e3f9c9ec9.png)

cristian-aldea commented on apps/mull-ui/src/app/pages/create-event/create-event.tsx:
```diff
@@ -0,0 +1,205 @@
+        <p className="create-event-text">Create Event</p>
+        <label htmlFor="imageFile" className="custom-file-upload event-input-border">
+          {imageFile ? (
+            <img src={imageFile} style={{ width: '50%', height: '50%' }} alt="sup" />
```
Please put something else for the alt text

JimLau49 replied:
Changing it to "Event" as Screen-readers already announce `img` tags as an image

cristian-aldea reviewed (changes requested):
Great work so far! I can confirm that creating events by filling out the form works and that form validation works. I can't also confirm that the backend works, and the event is saved to the database!

**Acceptance Criteria**

- [x] User acceptance flow is respected
- [ ] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [x] There must exist an automated Cypress acceptance test in GIF form (the video can be made into a gif afterwards, its no big deal)
- [ ] Must match [Mockup](create-mockup-link)

However, here are a few problems I found that I think should be addressed:

* ~~Currently the CI check is failing. From looking at the build it seems its caused by the fact that the backend doesn't run during the e2e test, you can run the backend as a separate background process and use wait-on, or use something like start-sever-and-test.~~ sorry wrote that before your fix commit. **However, running the backend as a background process can be finicky if it ends up taking a lot of time to come up. It might end up not being available before the e2e tests run. consider using something like `wait-on` or `start-server-and-test`**

* When I upload an image it looks weird on firefox and chrome, and there's huge padding
![image](https://user-images.githubusercontent.com/23544999/98137540-19171880-1e90-11eb-9026-0fad420dd1f7.png)

I propose the following styling:

create-event.tsx:
```tsx
<label htmlFor="imageFile" className="custom-file-upload event-input-border">
          {imageFile ? (
            <img
              src={imageFile}
              style={{ width: '100%', borderRadius: '20px', display: 'block' }}
              alt="sup"
            />
          ) : (
            <div style={{ margin: '1.8em 5em' }}>🧼</div>
          )}
        </label>
```

create-event.scss:

```scss
.custom-file-upload {
  display: block;
  cursor: pointer;

  margin-bottom: 20px;

  &:hover {
    background: #0ba024;
  }
}
```

Which results in this:

![image](https://user-images.githubusercontent.com/23544999/98137049-90987800-1e8f-11eb-90af-4b1e22328e5f.png)

* If end date isn't set, it shouldn't be an error. It should just be set to the start time.


* At certain resolutions, the time pickers are on the same line, and much wider than the rest of the form. In such situations, they could be on separate lines so as to not be wider than everything else.
![image](https://user-images.githubusercontent.com/23544999/98118677-92efd780-1e79-11eb-8ca2-c7d6e808b27d.png)

* **On desktop, the width of the calendar and input text fields don't match, but it does on the mockup. I see that the logic for setting the width of stuff on desktop is complex and hard-to-follow, and can be simplified. There's the `.page-container` class which sets the width to 50%. The calendar component then has a `.date-calendar` selector which sets the inner div to 65%. There's then `create-event-fields` which sets the width to 70% only on desktop. I recommend just keeping the `.page-container` and putting it to maybe 35%. If you do that you get this, which is simpler and I think looks better.**
![Screenshot from 2020-11-04 11-29-21](https://user-images.githubusercontent.com/23544999/98138459-1f59c480-1e91-11eb-8d36-89cf4f0e7b58.png)

* Selected option on the pill component is bold on the mockup, and I think that looks good. But it's not like that on the implementation

* There's a few instances of duplicate styling which could be cleaned up. One example are the scss selectors `.event-input-border` and `.input-border`

* Please add some documentation where you feel it might be needed to satisfy the AC :)

If you need help addressing some of these comments feel free to ask me for and I'd be glad to come and discuss, or walk you through what to do.

r-saba commented on apps/mull-ui/src/app/pages/create-event/create-event.tsx:
```diff
@@ -0,0 +1,205 @@
+        <p className="create-event-text">Create Event</p>
+        <label htmlFor="imageFile" className="custom-file-upload event-input-border">
+          {imageFile ? (
+            <img src={imageFile} style={{ width: '50%', height: '50%' }} alt="sup" />
```
Image shouldn't be in the `label` tag

cristian-aldea replied:
What should they do instead? I had a related comment

r-saba replied:
The label is being used here to be able to click the hidden input. What I've done in other projects is have a reference (with useRef hook) to the hidden input and when the parent div is clicked it calls the .click() function on the reference which is the input.

Since this PR doesn't deal with image uploading I'm okay with leaving the updating of the UI to #28. Just center the image to match the mockup.

cristian-aldea replied:
That's a good way of doing it yeah, if @JimLau49 and @bit172 don't want to tackle this now we can do it in #28. Feel free to use the code I suggested in my comment to center the image and remove the padding to better match the mockup.

r-saba commented on apps/mull-ui/src/app/app.tsx:
```diff
@@ -1,25 +1,37 @@
-import { Redirect, Route, Switch } from 'react-router-dom';
 import { ROUTES } from '../constants';
 
+import 'react-toastify/dist/ReactToastify.min.css';
```
Why do we have to explicitly import this? Never had to do this for a component I imported from a package. I see that's even how they do it in the docs, just leaving this comment to show my disdain.

TheGreatMarkus replied:
Actually that reminds me ragith, this import can be added to the styles array for the UI project in `workspace.json`

bit172 replied:
I think it has to be. 😕

r-saba commented on apps/mull-ui/src/app/pages/create-event/create-event.tsx:
```diff
@@ -0,0 +1,205 @@
+
+import './create-event.scss';
+
+/* eslint-disable-next-line */
```
We don't need this. I had no issues when I removed it locally.

bit172 replied:
The eslint line is just there when we generate components with Nx.

cristian-aldea replied:
Its because interfaces for props are already created for us with Nx, and they are initially empty. That line is there to avoid having tslint errors when you just generate components.

r-saba commented on libs/ui-lib/src/lib/custom-text-input/custom-text-input.tsx:
```diff
@@ -0,0 +1,42 @@
+
+import './custom-text-input.scss';
+
+/* eslint-disable-next-line */
```
Do we have this for a reason? I don't get any TS errors when I remove it and defining an interface shouldn't be an issue right?

r-saba commented on libs/ui-lib/src/lib/pill-options/pill-options.tsx:
```diff
@@ -0,0 +1,31 @@
+
+import './pill-options.scss';
+
+/* eslint-disable-next-line */
```
Do we need this?

r-saba commented on apps/mull-ui/src/app/pages/create-event/create-event.scss:
```diff
@@ -0,0 +1,90 @@
+  display: block;
+  padding: 1.8em 5em;
+  cursor: pointer;
+
```
Unecessary white space

r-saba commented on apps/mull-ui/src/app/pages/create-event/create-event.scss:
```diff
@@ -0,0 +1,90 @@
+  cursor: pointer;
+
+  margin-bottom: 20px;
+
```
Unecessary white space

r-saba commented on apps/mull-ui/src/app/pages/create-event/create-event.scss:
```diff
@@ -0,0 +1,90 @@
+    margin: 0 0 1.25rem;
+  }
+
+  @media (min-width: $breakpoint-mobile) {
+    flex-direction: row;
+    justify-content: space-evenly;
+    margin: 0 0 1.25rem;
+  }
```
Can you refactor it so that we have a single `@media (min-width: $breakpoint-mobile)` for the entire document. 

It will make the css more readable since if I want to look at desktop specific styles it will all be in one place.

r-saba commented on apps/mull-ui/src/app/pages/create-event/create-event.scss:
```diff
@@ -0,0 +1,90 @@
+.create-event-fields {
+  width: 100%;
+
+  @media (min-width: $breakpoint-mobile) {
+    width: 70%;
+  }
```
Move this into the single  `@media (min-width: $breakpoint-mobile)` I mentioned before.

example of what it should look like
```
@media (min-width: $breakpoint-mobile) {
  .create-event-fields {
    width: 70%;
  }
}
```

r-saba commented on apps/mull-api/src/app/event/event.service.ts:
```diff
@@ -0,0 +1,37 @@
+  }
+
+  async create(eventInput: CreateEventInput): Promise<Event> {
+    const newEvent = this.eventRepository.create(eventInput);
+    await this.eventRepository.save(newEvent);
+    return newEvent;
```
We don't need the .create() method.
```suggestion
    return await this.eventRepository.save(eventInput);
```

r-saba reviewed (changes requested):
Good work, left some comments. I've looked at code and functionality, haven't looked at how the UI compares to the mockup yet due to time. 

Also why do we have all these `istanbul ignores` now?

TheGreatMarkus commented:
> Also why do we have all these `istanbul ignores` now?

@r-saba I'll discuss it more during the scrum tomorrow, but it's because of how jest coverage works. The lines with anon functions aren't being considered covered, and adding that comment allows us to ignore those lines for coverage metrics.

leojrsilao commented on apps/mull-api/src/app/event/inputs/event.input.ts:
```diff
@@ -0,0 +1,47 @@
+  @Field(/* istanbul ignore next */ () => Int)
+  restriction: EventRestriction;
+
+  //   host: User;
+  // image: Media;
+  // location: Location;
```
Can remove these comments if they are not needed

JimLau49 replied:
We're gonna leave these as TODO's to so that the developers taking #28 will be able to uncomment these types and use them

leojrsilao commented on apps/mull-api/src/app/event/inputs/event.input.ts:
```diff
@@ -0,0 +1,47 @@
+
+  @Field(/* istanbul ignore next */ () => Int, { nullable: true })
+  restriction: EventRestriction;
+  // image: Media;
+  // location: Location;
```
Can remove these comments if they are not needed

leojrsilao commented on libs/ui-lib/src/lib/custom-text-input/custom-text-input.scss:
```diff
@@ -0,0 +1,23 @@
+
+.input-border {
+  border-radius: 20px;
+  border: 1.5px #27b09a solid;
```
Can change the #27b09a to $primary-color to keep it consistent with our variables.scss constants

JimLau49 replied:
Will do 🚀

cristian-aldea replied:
yes, and please do this with all colors in the styling for the app

JimLau49 replied:
Done

leojrsilao commented on libs/ui-lib/src/lib/custom-time-picker/custom-time-picker.scss:
```diff
@@ -0,0 +1,50 @@
+  }
+
+  &::-webkit-datetime-edit-ampm-field {
+    background-color: #27b09a;
```
Same as previous comment

leojrsilao commented on libs/ui-lib/src/lib/pill-options/pill-options.scss:
```diff
@@ -0,0 +1,27 @@
+}
+
+.pill-option {
+  border: 1.3px #27b09a solid;
```
Same as the previous comment

leojrsilao commented on apps/mull-ui/src/app/pages/create-event/create-event.scss:
```diff
@@ -0,0 +1,90 @@
+
+.event-input-border {
+  border-radius: 20px;
+  border: 1.5px #27b09a solid;
```
Same as previous comment

leojrsilao commented on libs/ui-lib/src/lib/pill-options/pill-options.scss:
```diff
@@ -0,0 +1,27 @@
+  }
+
+  &.active {
+    background-color: #27b09a;
```
Same as previous comment

leojrsilao reviewed (changes requested):
Good work! Noticed that some UI text boxes are missing with their appropriate icons and the SVG icon for the upload photo is not the same as our mockups, left minor comments to be changed (nothing harmful to functionality), everything else LGTM! 🚀 🚀 

**Figma Mockup:**
![image](https://user-images.githubusercontent.com/43038259/98182051-cad73900-1ed2-11eb-9264-281cadcec131.png)
![image](https://user-images.githubusercontent.com/43038259/98182296-6f597b00-1ed3-11eb-8a15-bcdbeb6af60b.png)
======
**Current UI:**
![image](https://user-images.githubusercontent.com/43038259/98182120-fc500480-1ed2-11eb-96a3-9c7b213fe6ed.png)
![image](https://user-images.githubusercontent.com/43038259/98182256-55b83380-1ed3-11eb-9cef-0f3f4f4e6028.png)

- [x] User acceptance flow is respected
- [ ] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [x] There must exist an automated Cypress acceptance test in GIF form
- [ ] Must match [Mockup](create-mockup-link)

bit172 committed:
Moved styling to ui-lib #27
This allows ui components to use the same styling as the react app

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>
Co-authored-by: Cristian Aldea <aldea.cristian@outlook.com>

bit172 committed:
Address PR comments #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

r-saba committed:
Add svg icons to create event form #27

r-saba commented:
As we discussed, I quickly added and positioned the svg icons to the form to help resolve @leojrsilao pr comments and to allow @JimLau49 @bit172 to focus on the other comments. I will stay on as a reviewer since I was not involved in the rest of the PR and would like to make sure my comments above get addressed.

JimLau49 commented:
> As we discussed, I quickly added and positioned the svg icons to the form to help resolve @leojrsilao pr comments and to allow @JimLau49 @bit172 to focus on the other comments. I will stay on as a reviewer since I was not involved in the rest of the PR and would like to make sure my comments above get addressed.

Thanks for your hard work @r-saba

bit172 committed:
Address PR comments #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

bit172 committed:
Run mull-api in travis #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>
Co-authored-by: Ritchelle Posadas  <Ritchelle.posadas@gmail.com>

bit172 committed:
Add documentation #27

Co-authored-by: Jimmy Lau <jimmy71471@gmail.com>

cristian-aldea commented on libs/ui-lib/src/lib/pill-options/pill-options.spec.tsx:
```diff
@@ -0,0 +1,44 @@
+import PillOptions from './pill-options';
+
+const mockOptions: string[] = ['Yes', 'No', 'Maybe'];
+const mockOnChange: jest.Mock<any, any> = jest.fn();
```
The use of any gives a linting warning, you can just remove the type and the test still passes:
```suggestion
const mockOnChange: jest.Mock = jest.fn();
```

cristian-aldea commented on apps/mull-ui/src/app/pages/create-event/create-event.tsx:
```diff
@@ -0,0 +1,238 @@
+
+    validationSchema: Yup.object({
+      startDate: Yup.date().required('Start date is required'),
+      endDate: Yup.date()
+        .nullable()
+        .required('End date is required')
```
I don't think that having a null endDate should be an error. If the user didn't set the endDate, I think its safe to assume that they want a one-day event.

r-saba replied:
Agreed, choosing a single date wasn't intuitive because I had to tap the same date twice.

bit172 replied:
I left it as nullable and will set it as the start date in the submit logic.

cristian-aldea commented on apps/mull-ui/src/app/pages/create-event/create-event.scss:
```diff
@@ -0,0 +1,80 @@
+
+.event-input-border {
+  border-radius: 8px;
+  border: 1.5px $primary-color solid;
+  &.error {
+    border-color: #ff0000;
+  }
```
See `libs/ui-lib/src/lib/custom-text-input/custom-text-input.scss`. This style block is a duplicate and can be moved to a mixin or separate class in ui-lib.

cristian-aldea commented on apps/mull-ui/src/app/pages/create-event/create-event.scss:
```diff
@@ -0,0 +1,80 @@
+}
+
+@media (min-width: $breakpoint-mobile) {
+  .create-event {
+    padding: 1rem 0 0;
+  }
```
Text is too small on desktop
Mockup:
![image](https://user-images.githubusercontent.com/23544999/98428389-3fdf7580-206f-11eb-9ab9-37f998dc3301.png)
Implementation:
![image](https://user-images.githubusercontent.com/23544999/98428408-5c7bad80-206f-11eb-9f10-8ec9105c1662.png)

```suggestion
    .create-event {
    padding: 1rem 0 0;
  }
  .create-event-text {
    font-size: 1.5rem;
  }
```

r-saba reviewed (changes requested):
It's looking really good. Can you make it so we only have to click once on the calendar to make a one day event.  Having it so that end date is not required.

leojrsilao reviewed (changes requested):
Looking better now! Noticed a little bug, in the `.custom-text-input`, just have to add 
`padding-right: 2.5rem` or whichever rem value that doesn't overflow the text with the SVG icons. Other than that LGTM! 🚀 
![image](https://user-images.githubusercontent.com/43038259/98429794-e11dfa00-2076-11eb-8df6-cd952526f89e.png)

cristian-aldea committed:
Refactor ci script to use wait-on module

Also, refactored code out of script into seperate files to make team easy to run locally for testing

TheGreatMarkus commented:
So @JimLau49  @bit172 

I saw you didn't address one of the comments related to the ci for the e2e tests, I created a branch called [cris-ci-suggestion](https://github.com/RGPosadas/Mull/tree/US-1.1/cris-ci-suggestion) with a commit to fix it and use `wait-on`. The build passed, so have a look at it and lemme know what you think.

JimLau49 commented:
> So @JimLau49 @bit172
> 
> I saw you didn't address one of the comments related to the ci for the e2e tests, I created a branch called [cris-ci-suggestion](https://github.com/RGPosadas/Mull/tree/US-1.1/cris-ci-suggestion) with a commit to fix it and use `wait-on`. The build passed, so have a look at it and lemme know what you think.

Hey cristian, yeah this looks good to me from my understanding, we can include it to this PR

JimLau49 commented:
> It's looking really good. Can you make it so we only have to click once on the calendar to make a one day event. Having it so that end date is not required.

alright consider it done

JimLau49 committed:
address pr comments #27

Co-authored by: Jonathan Hsu <bit172@gmail.com>

TheGreatMarkus commented:
I added my commit from [cris-ci-suggestion](https://github.com/RGPosadas/Mull/tree/US-1.1/cris-ci-suggestion) with @JimLau49 and @bit172 's approval to help them out, since they were busy with addressing other comments, and weren't too familiar with how to adjust the ci

leojrsilao reviewed (approved):
Everything LGTM!! 🚀 🚀 Just another minor UI change that can be done, but will approve, the preview image isn't centered on the container, can add in the following to center it: 

**In the .custom-file-upload:** 
```
 text-align: center;
```

Image not centered: 
![image](https://user-images.githubusercontent.com/43038259/98452669-7b397d00-211f-11eb-8269-c75965d89c31.png)

cristian-aldea reviewed (approved):
- [x] User acceptance flow is respected
- [x] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [x] There must exist an automated Cypress acceptance test in GIF form (will be done later)
- [x] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

Thanks for addressing all the comments, looks good to me!

TheGreatMarkus commented:
> Everything LGTM!! 🚀 🚀 Just another minor UI change that can be done, but will approve, the preview image isn't centered on the container, can add in the following to center it:
> 
> **In the .custom-file-upload:**
> 
> ```
>  text-align: center;
> ```
> 
> Image not centered:
> ![image](https://user-images.githubusercontent.com/43038259/98452669-7b397d00-211f-11eb-8269-c75965d89c31.png)

We discussed about this, it will be addressed in #28

r-saba reviewed (approved):
Thanks for your hardwork. LGTM!
**Acceptance Criteria**

- [X] User acceptance flow is respected
- [X] Documentation present for every new component and function
- [X] Unit and UI tests are implemented for every component and function
- [X] There must exist an automated Cypress acceptance test in GIF form (Done later)
- [X] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528) (Mockups will be updated to match this PR)
